casbah/src/main/scala/akka/contrib/persistence/mongodb/CasbahPersistenceJournaller.scala

```diff
@@ -45,49 +45,33 @@ class CasbahPersistenceJournaller(driver: CasbahMongoDriver) extends MongoPersis
       .map(driver.deserializeJournal)
   }
 
-  import collection.immutable.{ Seq => ISeq }
-  private[this] def doBatchJournalAppend(writes: ISeq[AtomicWrite], journal: MongoCollection)(implicit ec: ExecutionContext): ISeq[Try[Unit]] = {
+  import collection.immutable.{ Seq => ISeq }  
+  private[this] def doBatchAppend(writes: ISeq[AtomicWrite], collection: MongoCollection)(implicit ec: ExecutionContext): ISeq[Try[Unit]] = {
     val batch = writes.map(write => Try(driver.serializeJournal(Atom[DBObject](write, driver.useLegacySerialization))))
     
     if (batch.forall(_.isSuccess)) {
-      val bulk = journal.initializeOrderedBulkOperation
+      val bulk = collection.initializeOrderedBulkOperation
       batch.collect { case scala.util.Success(ser) => ser } foreach bulk.insert
       bulk.execute(writeConcern)
       batch.map(t => t.map(_ => ()))
     } else { // degraded performance, can't batch
-      batch.map(_.map { serialized =>
-        journal.insert(serialized)(identity, writeConcern)
-        //if (driver.realtimeEnablePersistence) realtime.insert(serialized)(identity, writeConcern)
-      }.map(_ => ()))
-    }
-  }
-
-  private[this] def doBatchRealtimeAppend(writes: ISeq[AtomicWrite])(implicit ec: ExecutionContext): ISeq[Try[Unit]] = {
-    val batch = writes.map(write => Try(driver.serializeJournal(Atom[DBObject](write, driver.useLegacySerialization))))
-    if (batch.forall(_.isSuccess)) {
-      val bulk = realtime.initializeOrderedBulkOperation
-      batch.collect { case scala.util.Success(ser) => ser } foreach bulk.insert
-      bulk.execute(writeConcern)
-
-      batch.map(t => t.map(_ => ()))
-    } else { // degraded performance, can't batch
-      batch.map(_.map { serialized => realtime.insert(serialized)(identity, writeConcern) }.map(_ => ()))
+      batch.map(_.map { serialized => collection.insert(serialized)(identity, writeConcern) }.map(_ => ()))
     }
   }
 
   private[mongodb] override def batchAppend(writes: ISeq[AtomicWrite])(implicit ec: ExecutionContext): Future[ISeq[Try[Unit]]] = {
     val batchFuture = Future {
       if (driver.useSuffixedCollectionNames) {
-        writes.groupBy(_.persistenceId).flatMap {
-          case (pid, writeSeq) => doBatchJournalAppend(writeSeq, driver.journal(pid))
+        writes.groupBy(w => driver.getSuffixFromPersistenceId(w.persistenceId)).flatMap {
+          case (pid, writeSeq) => doBatchAppend(writeSeq, driver.journal(pid))
```

Comment:
Can you rename `pid` to `suffix` - also does `driver.journal` work when passed the suffix?